src/components/Frame/components/Loading/Loading.tsx

```diff
@@ -1,79 +1,47 @@
-import React, {Component} from 'react';
-import debounce from 'lodash/debounce';
+import React, {useEffect, useState} from 'react';
 
-import styles from './Loading.scss';
-
-export interface LoadingProps {}
+import {useI18n} from '../../../../utilities/i18n';
 
-interface State {
-  progress: number;
-  step: number;
-  animation: number | null;
-}
+import styles from './Loading.scss';
 
-const INITIAL_STEP = 10;
 const STUCK_THRESHOLD = 99;
 
-export class Loading extends Component<LoadingProps, State> {
-  state: State = {
-    progress: 0,
-    step: INITIAL_STEP,
-    animation: null,
-  };
-
-  private ariaValuenow = debounce(() => {
-    const {progress} = this.state;
-    return Math.floor(progress / 10) * 10;
-  }, 15);
-
-  componentDidMount() {
-    this.increment();
-  }
-
-  componentWillUnmount() {
-    const {animation} = this.state;
-
-    if (animation != null) {
-      cancelAnimationFrame(animation);
-    }
-  }
-
-  render() {
-    const {progress} = this.state;
-
-    const customStyles = {
-      transform: `scaleX(${Math.floor(progress) / 100})`,
-    };
-
-    const ariaValuenow = this.ariaValuenow();
-
-    return (
-      <div className={styles.Loading}>
-        <div
-          className={styles.Level}
-          style={customStyles}
-          aria-valuenow={ariaValuenow}
-          aria-valuemin={0}
-          aria-valuemax={100}
-          role="progressbar"
-        />
-      </div>
-    );
-  }
+export function Loading() {
+  const i18n = useI18n();
 
-  private increment() {
-    const {progress, step} = this.state;
+  const [progress, setProgress] = useState(0);
+  const [animating, setAnimating] = useState(false);
 
-    if (progress >= STUCK_THRESHOLD) {
+  useEffect(() => {
+    if (progress >= STUCK_THRESHOLD || animating) {
       return;
     }
 
-    const animation = requestAnimationFrame(() => this.increment());
-
-    this.setState({
-      progress: Math.min(progress + step, 100),
-      step: INITIAL_STEP ** -(progress / 25),
-      animation,
+    requestAnimationFrame(() => {
+      const step = Math.max((STUCK_THRESHOLD - progress) / 10, 1);
+      setAnimating(true);
+      setProgress(progress + step);
     });
-  }
+  }, [progress, animating]);
+
+  const customStyles = {
+    transform: `scaleX(${Math.floor(progress) / 100})`,
+  };
+
+  return (
+    <div
+      className={styles.Loading}
+      aria-valuenow={progress}
```

Comment:
Taking a read of https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_aria-valuenow_attribute

> When the rendered value cannot be accurately represented as a number, authors SHOULD use the aria-valuetext attribute in conjunction with aria-valuenow to provide a user-friendly representation of the range's current value. For example, a slider may have rendered values of small, medium, and large. In this case, the values of aria-valuenow could range from 1 through 3, which indicate the position of each value in the value space, but the aria-valuetext would be one of the strings: small, medium, or large.


Seeing as we're just increasing the percentage over time rather than being in relation to anything meaningful, I'd argue that loading does not have accurate numerical representation.

If that's the case, what if we stop using`aria-valuenow` (and aria-valuemin / aria-valuemax`) and instead set `aria-valuetext="Loading in progress"` or some other not-pretending-there-is-an-exact-value text. Then we'd be able to move to the CSS animation approach for the visual effect.

Reply:
Yeah I agree, but lets make a new issue or PR to do that. That is outside of the scope of this fix.